indexing-service/src/main/java/org/apache/druid/indexing/worker/shuffle/LocalIntermediaryDataManager.java

```diff
@@ -377,20 +377,14 @@ public DataSegment addSegment(String supervisorTaskId, String subTaskId, DataSeg
       final File partitionDir = new File(location.getPath(), getPartitionDirPath(supervisorTaskId, interval, bucketId));
       if (partitionDir.exists()) {
         supervisorTaskCheckTimes.put(supervisorTaskId, getExpiryTimeFromNow());
-        final File[] segmentFiles = partitionDir.listFiles();
-        if (segmentFiles == null) {
-          return Optional.empty();
+        final File segmentFile = new File(partitionDir, subTaskId);
```

Comment:
LGTM is complaining because `subTaskId` is passed in from user input. A validation check similar to the one done for `supervisorTaskId` on line 375 should suppress the warning from LGTM

Reply:
Thanks, fixed

Reply:
The new code is semantically the same, with the same branching (i.e. partition found it returns it else it returns Optional.empty() so I think we are good. The Optional.empty() return though is not being hit by any existing unit test.

Reply:
It looks like the existing unit tests don't validate what happens when the segment file is missing.
<img width="1112" alt="Screen Shot 2021-11-11 at 9 38 30 AM" src="https://user-images.githubusercontent.com/44787917/141343789-f575c851-a810-4bb8-8cbc-8fc25a0faee4.png">

This logic is simple enough, and code coverage bot is happy, so I'm ok with not adding a test case for this if it seems like overkill